WcaOnRails/app/models/user.rb

```diff
@@ -632,15 +632,33 @@ def banned?
     current_teams.include?(Team.banned)
   end
 
+  def ban_end
+    current_team_members.where(team: Team.banned).first.end_date
+  end
+
   def banned_at_date?(date)
     if banned?
-      ban_end = current_team_members.select(:team == Team.banned).first.end_date
       !ban_end.present? || date < ban_end
     else
       false
     end
   end
 
+  def permissions
+    {
+      can_attend_competitions: {
+        scope: cannot_register_for_competition_reasons.empty? ? "*" : [],
+        until: banned? ? ban_end : nil,
```

Comment:
Wouldn't it make more sense to straight-out discard this key if the user is banned?

Reply:
Also, what is the overlap between `cannot_register_for_competition_reasons` and `banned?`? Are we confident that the first returns something non-empty exactly if (genau dann, wenn) the other is not empty?

Even if we are confident about that, why are we using both checks disjointly?

Reply:
`cannot_register_for_competition_reasons` is not empty if a user is banned OR if they have not set up their DOB/name/gender/country. I agree that setting until only if banned makes sense, can I do that directly in the has somehow?